src/error.rs

```diff
@@ -0,0 +1,28 @@
+#[derive(thiserror::Error, Debug)]
+pub enum GPUError {
+    #[error("Ocl Error: {0}")]
+    Ocl(ocl::Error),
```

Comment:
I think you can just write `#[from] ocl::Error` instead, and have it derive the from impl for you